src/xdsl/xdsl_opt_main.py

```diff
@@ -66,8 +67,33 @@ def run(self):
         else:
             try:
                 self.apply_passes(module)
+
             except DiagnosticException as e:
-                print(e)
+
+                def gen_frame(e: DiagnosticException, n: int) -> Frame:
+                    return Frame(e, n) if n >= 1 else Frame(e, 3)
+
+                def verbose_frame_output(f: Frame, m: int) -> None:
+                    f.verbose(m) if m > 1 else f.verbose(1)
```

Comment:
I think it would be cool to have an argument for the number of frames printed that just prints the entire stack trace in this form. Maybe this could be one of the functionalities with less arguments?

Reply:
Sorry, I don't think I got what do you mean by that. Do you mean printing out the entire traceback in that format? If that's the case, I have tried it and most of them are not very helpful

Reply:
Yes, since I usually want to know the entire call stack in order to understand the error.

Why do you think it's not helpful? Too much information?

Reply:
Yes, sometimes it gets overly verbose to read with too much information. But I can see that being an option to the user, when they just want to print everything all at once.

Reply:
Okay, yes I see that this could be the case. Still, I feel that the usual flow will be:
1. run: oh it crashed
2. run: run it with verbose traceback and just some numbers
3. run: (if 2. was not enough info) print the entire stack trace to see everything.

What do you think?

Reply:
I changed the meaning of num of frames:
```
      num:
          if num > 0: last num frames
          if num == 0: all frames
          if num < 0: drop first num frames, get all the remaining frames
```
@webmiche is this what you want?

Reply:
Yes, I like this! Now it just needs to be indicated in the help message :)